src/common/HttpRequest.cc

```diff
@@ -0,0 +1,156 @@
+/* not use this file except in compliance with the License. You may obtain    */
+/* a copy of the License at                                                   */
+/*                                                                            */
+/* http://www.apache.org/licenses/LICENSE-2.0                                 */
+/*                                                                            */
+/* Unless required by applicable law or agreed to in writing, software        */
+/* distributed under the License is distributed on an "AS IS" BASIS,          */
+/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
+/* See the License for the specific language governing permissions and        */
+/* limitations under the License.                                             */
+/* -------------------------------------------------------------------------- */
+
+#include "HttpRequest.h"
+#include <iostream>
+
+using namespace std;
+
+
+HttpRequest::HttpRequest()
+{
+    curl_global_init(CURL_GLOBAL_ALL);
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+HttpRequest::~HttpRequest()
+{
+    curl_global_cleanup();
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+int HttpRequest::post_json(const std::string& url, const std::string& data, std::string& response)
+{
+    auto curl = curl_easy_init();
+
+    if (!curl) return -1;
+
+    auto headers = curl_slist_append(nullptr, "Accept: application/json");
+    headers = curl_slist_append(headers, "Content-Type: application/json");
+
+    curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+    curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+    curl_easy_setopt(curl, CURLOPT_POSTFIELDS, data.c_str());
+    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_to_string);
+    curl_easy_setopt(curl, CURLOPT_WRITEDATA, &response);
+
+    if (_timeout != 0)
+    {
+        curl_easy_setopt(curl, CURLOPT_TIMEOUT, _timeout);
+    }
+
+    if (!_proxy.empty())
+    {
+        curl_easy_setopt(curl, CURLOPT_PROXY, _proxy);
+    }
+
+    auto ec = curl_easy_perform(curl);
+
+    if (ec != CURLE_OK)
+    {
+        response = curl_easy_strerror(ec);
+
+        curl_slist_free_all(headers);
+        curl_easy_cleanup(curl);
+
+        return -1;
+    }
+
+    auto rc = check_http_code(curl, response);
+
+    curl_slist_free_all(headers);
+    curl_easy_cleanup(curl);
+
+    return rc;
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+int HttpRequest::get_json(const std::string& url, std::string& response)
+{
+    auto curl = curl_easy_init(); // todo use RAII to automatically clean up
+
+    if (!curl) return -1;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that in these codes. `Multiline blocks` are missing. It is considered best practice to enclose all `multiline blocks` including if else blocks, loops, and function bodies within curly braces. To mitigate this issue, it serves to clearly define the scope of the block and avoid logic errors by grouping all related statements together visually.